src/block/Hopper.php

```diff
@@ -92,7 +101,80 @@ public function onInteract(Item $item, int $face, Vector3 $clickVector, ?Player
 	}
 
 	public function onScheduledUpdate() : void{
-		//TODO
+		$world = $this->position->getWorld();
+
+		$hopperBlock = $this->position->getWorld()->getBlock($this->position);
+		if(!$hopperBlock instanceof Hopper){
+			return;
+		}
+
+		if($this->hopperOnCooldown()){
+			$this->currentCooldown--;
```

Comment:
I think that rather than doing it this way, it might be better to store the time of the last interaction to allow verification.  

Thus, the hopper always updates at 8 tick when an action is performed (and updates without worrying about cooldown) and if, when updating, the difference isn't greater than the cooldown, we don't go any further.